### PR TITLE
UCP/RECV/UNPACK: drop data on failed receive

### PR DESCRIPTION
**Pull request by hoopoepg**
- in case if receive operation is failed - just drop incoming data

### CODE REVIEW

**Review by dmitrygx — DISMISSED** [review]

File: src/ucp/core/ucp_request.inl [hunk 1]
```diff
@@ -570,10 +570,16 @@ ucp_request_recv_data_unpack(ucp_request_t *req, const void *data,
...
+                  "unpack recv_data req_len %zu data_len %zu offset %zu "
+                  "last %s status %s",
+                  req->recv.length, length, offset, last ? "yes" : "no",
+                  ucs_status_string(req->status));
 
-    ucp_trace_req(req, "unpack recv_data req_len %zu data_len %zu offset %zu last: %s",
-                  req->recv.length, length, offset, last ? "yes" : "no");
+    if (ucs_unlikely(req->status != UCS_OK)) {
+        /* in case if status is not OK - just return */
+        return req->status;
```
Comment (brminich) [inline src/ucp/core/ucp_request.inl:581]: could this fail an assert in `uct_iface_invoke_am`, which expects either `OK` or `INPROGRESS`?

File: src/ucp/core/ucp_request.inl [hunk 2]
```diff
@@ -570,10 +570,16 @@ ucp_request_recv_data_unpack(ucp_request_t *req, const void *data,
...
+    ucp_trace_req(req,
+                  "unpack recv_data req_len %zu data_len %zu offset %zu "
+                  "last %s status %s",
+                  req->recv.length, length, offset, last ? "yes" : "no",
+                  ucs_status_string(req->status));
 
-    ucp_trace_req(req, "unpack recv_data req_len %zu data_len %zu offset %zu last: %s",
-                  req->recv.length, length, offset, last ? "yes" : "no");
+    if (ucs_unlikely(req->status != UCS_OK)) {
+        /* in case if status is not OK - just return */
```
Comment (brminich) [inline src/ucp/core/ucp_request.inl:580]: please remove this comment, it just duplicates the code

File: src/ucp/core/ucp_request.inl (hunk 1 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_request.inl:580]: evaluation updated

Comment (brminich) [reply src/ucp/core/ucp_request.inl:580]: this func still can return non `OK` or `INPROGRESS`

Comment (hoopoepg) [reply src/ucp/core/ucp_request.inl:580]: right, this is correct behavior

File: src/ucp/tag/eager_rcv.c [hunk 3]
```diff
@@ -107,10 +107,9 @@ ucp_eager_tagged_handler(void *arg, void *data, size_t length, unsigned am_flags
             req->recv.tag.info.length =
             req->recv.remaining       = eagerf_hdr->total_len;
 
-            status = ucp_tag_request_process_recv_data(req,
-                                                       UCS_PTR_BYTE_OFFSET(data, hdr_len),
-                                                       recv_len, 0, 0, flags);
-            ucs_assert((status == UCS_OK) || (status == UCS_INPROGRESS));
```
Comment (brminich) [inline src/ucp/tag/eager_rcv.c:113]: 👍

File: src/ucp/core/ucp_request.inl (hunk 1 above)
Comment (yosefe) [reply src/ucp/core/ucp_request.inl:580]: this cond is already checked by ucp_request_process_recv_data()
i think we can just remove the assert from eager_rcv.c:113

Comment (hoopoepg) [reply src/ucp/core/ucp_request.inl:580]: done

Comment (yosefe) [reply src/ucp/core/ucp_request.inl:580]: > done

not pushed..

Comment (brminich) [reply src/ucp/core/ucp_request.inl:580]: it is already there

**Comment by hoopoepg** [comment]
replaced by https://github.com/openucx/ucx/pull/6599

**Pull request closed without merging**